packages/check/match.js

```diff
@@ -115,7 +115,9 @@ var Optional = function (pattern) {
   this.pattern = pattern;
 };
 
-var Maybe = Optional;
+var Maybe = function (pattern) {
+  this.pattern = pattern;
+};
 
```

Comment:
Line 256 below is where the real problem happened since it's an `elseif` that never got reached anymore since both `.Maybe` and `.Optional` were the same object.